ElasticsearchTest

```
----------------------------------------
Cluster Name: elasticsearch-cluster-runner
Base Path:    /tmp/es-cluster431932489297553471
----------------------------------------
Cluster Name: elasticsearch-cluster-runner
Base Path:    /tmp/es-cluster4836132851537540940
Num Of Node:  1
----------------------------------------
Num Of Node:  1
----------------------------------------
Creating /tmp/es-cluster431932489297553471/config/node_1
Creating /tmp/es-cluster4836132851537540940/config/node_1
Creating /tmp/es-cluster431932489297553471/logs/node_1
Creating /tmp/es-cluster431932489297553471/data/node_1
Creating /tmp/es-cluster4836132851537540940/logs/node_1
Creating /tmp/es-cluster4836132851537540940/data/node_1
Node Name:      Node 1
HTTP Port:      9211
Transport Port: 9311
Data Directory: /tmp/es-cluster4836132851537540940/data/node_1
Log Directory:  /tmp/es-cluster4836132851537540940/logs/node_1
----------------------------------------
Node Name:      Node 1
HTTP Port:      9201
Transport Port: 9301
Data Directory: /tmp/es-cluster431932489297553471/data/node_1
Log Directory:  /tmp/es-cluster431932489297553471/logs/node_1
----------------------------------------
Creating /tmp/es-cluster431932489297553471/modules
Creating /tmp/es-cluster431932489297553471/plugins
Creating /tmp/es-cluster4836132851537540940/modules
Creating /tmp/es-cluster4836132851537540940/plugins
[2017-09-19T11:38:00,030][INFO ][o.e.n.Node               ] [Node 1] initializing ...
[2017-09-19T11:38:00,029][INFO ][o.e.n.Node               ] [Node 1] initializing ...
[2017-09-19T11:38:00,174][INFO ][o.e.e.NodeEnvironment    ] [Node 1] using [1] data paths, mounts [[/ (/dev/sda1)]], net usable_space [16gb], net total_space [29.4gb], spins? [no], types [ext4]
[2017-09-19T11:38:00,179][INFO ][o.e.e.NodeEnvironment    ] [Node 1] using [1] data paths, mounts [[/ (/dev/sda1)]], net usable_space [16gb], net total_space [29.4gb], spins? [no], types [ext4]
[2017-09-19T11:38:00,189][INFO ][o.e.e.NodeEnvironment    ] [Node 1] heap size [1007.3mb], compressed ordinary object pointers [true]
[2017-09-19T11:38:00,192][INFO ][o.e.n.Node               ] [Node 1] node name [Node 1], node ID [0uo6S3NHSzihlcDfJqJ_-g]
[2017-09-19T11:38:00,194][INFO ][o.e.n.Node               ] [Node 1] version[5.6.0], pid[7311], build[781a835/2017-09-07T03:09:58.087Z], OS[Linux/4.4.0-93-generic/amd64], JVM[Oracle Corporation/Java HotSpot(TM) 64-Bit Server VM/1.8.0_144/25.144-b01]
[2017-09-19T11:38:00,198][INFO ][o.e.n.Node               ] [Node 1] JVM arguments [-Dfile.encoding=UTF8, -Xms1g, -Xmx1g, -Xss6M, -XX:+CMSClassUnloadingEnabled, -XX:+UseConcMarkSweepGC, -XX:ReservedCodeCacheSize=128m]
[2017-09-19T11:38:00,203][INFO ][o.e.e.NodeEnvironment    ] [Node 1] heap size [1007.3mb], compressed ordinary object pointers [true]
[2017-09-19T11:38:00,207][INFO ][o.e.n.Node               ] [Node 1] node name [Node 1], node ID [8SwMlrshQgqYibIM7hsSVw]
[2017-09-19T11:38:00,207][INFO ][o.e.n.Node               ] [Node 1] version[5.6.0], pid[7311], build[781a835/2017-09-07T03:09:58.087Z], OS[Linux/4.4.0-93-generic/amd64], JVM[Oracle Corporation/Java HotSpot(TM) 64-Bit Server VM/1.8.0_144/25.144-b01]
[2017-09-19T11:38:00,207][INFO ][o.e.n.Node               ] [Node 1] JVM arguments [-Dfile.encoding=UTF8, -Xms1g, -Xmx1g, -Xss6M, -XX:+CMSClassUnloadingEnabled, -XX:+UseConcMarkSweepGC, -XX:ReservedCodeCacheSize=128m]
[2017-09-19T11:38:00,231][INFO ][o.e.p.PluginsService     ] [Node 1] no modules loaded
[2017-09-19T11:38:00,232][INFO ][o.e.p.PluginsService     ] [Node 1] no modules loaded
[2017-09-19T11:38:00,235][INFO ][o.e.p.PluginsService     ] [Node 1] loaded plugin [org.elasticsearch.index.reindex.ReindexPlugin]
[2017-09-19T11:38:00,237][INFO ][o.e.p.PluginsService     ] [Node 1] loaded plugin [org.elasticsearch.ingest.common.IngestCommonPlugin]
[2017-09-19T11:38:00,239][INFO ][o.e.p.PluginsService     ] [Node 1] loaded plugin [org.elasticsearch.painless.PainlessPlugin]
[2017-09-19T11:38:00,241][INFO ][o.e.p.PluginsService     ] [Node 1] loaded plugin [org.elasticsearch.percolator.PercolatorPlugin]
[2017-09-19T11:38:00,244][INFO ][o.e.p.PluginsService     ] [Node 1] loaded plugin [org.elasticsearch.script.expression.ExpressionPlugin]
[2017-09-19T11:38:00,247][INFO ][o.e.p.PluginsService     ] [Node 1] loaded plugin [org.elasticsearch.script.groovy.GroovyPlugin]
[2017-09-19T11:38:00,247][INFO ][o.e.p.PluginsService     ] [Node 1] loaded plugin [org.elasticsearch.script.mustache.MustachePlugin]
[2017-09-19T11:38:00,247][INFO ][o.e.p.PluginsService     ] [Node 1] loaded plugin [org.elasticsearch.search.aggregations.matrix.MatrixAggregationPlugin]
[2017-09-19T11:38:00,247][INFO ][o.e.p.PluginsService     ] [Node 1] loaded plugin [org.elasticsearch.transport.Netty3Plugin]
[2017-09-19T11:38:00,247][INFO ][o.e.p.PluginsService     ] [Node 1] loaded plugin [org.elasticsearch.transport.Netty4Plugin]
[2017-09-19T11:38:00,248][INFO ][o.e.p.PluginsService     ] [Node 1] loaded plugin [org.elasticsearch.index.reindex.ReindexPlugin]
[2017-09-19T11:38:00,248][INFO ][o.e.p.PluginsService     ] [Node 1] loaded plugin [org.elasticsearch.ingest.common.IngestCommonPlugin]
[2017-09-19T11:38:00,249][INFO ][o.e.p.PluginsService     ] [Node 1] loaded plugin [org.elasticsearch.painless.PainlessPlugin]
[2017-09-19T11:38:00,249][INFO ][o.e.p.PluginsService     ] [Node 1] loaded plugin [org.elasticsearch.percolator.PercolatorPlugin]
[2017-09-19T11:38:00,249][INFO ][o.e.p.PluginsService     ] [Node 1] loaded plugin [org.elasticsearch.script.expression.ExpressionPlugin]
[2017-09-19T11:38:00,254][INFO ][o.e.p.PluginsService     ] [Node 1] loaded plugin [org.elasticsearch.script.groovy.GroovyPlugin]
[2017-09-19T11:38:00,255][INFO ][o.e.p.PluginsService     ] [Node 1] loaded plugin [org.elasticsearch.script.mustache.MustachePlugin]
[2017-09-19T11:38:00,255][INFO ][o.e.p.PluginsService     ] [Node 1] loaded plugin [org.elasticsearch.search.aggregations.matrix.MatrixAggregationPlugin]
[2017-09-19T11:38:00,257][INFO ][o.e.p.PluginsService     ] [Node 1] loaded plugin [org.elasticsearch.transport.Netty3Plugin]
[2017-09-19T11:38:00,257][INFO ][o.e.p.PluginsService     ] [Node 1] loaded plugin [org.elasticsearch.transport.Netty4Plugin]
Closed all nodes.
Deleted /tmp/es-cluster4836132851537540940
[info] Test run started
[error] Test akka.stream.alpakka.elasticsearch.ElasticsearchTest failed: org.codelibs.elasticsearch.runner.ClusterRunnerException: Failed to start node 1, took 0.0 sec
[error]     at org.codelibs.elasticsearch.runner.ElasticsearchClusterRunner.build(ElasticsearchClusterRunner.java:360)
[error]     at org.codelibs.elasticsearch.runner.ElasticsearchClusterRunner.build(ElasticsearchClusterRunner.java:284)
[error]     at akka.stream.alpakka.elasticsearch.ElasticsearchTest.setup(ElasticsearchTest.java:43)
[error]     ...
[error] Caused by: java.lang.IllegalStateException: availableProcessors is already set to [2], rejecting [2]
[error]     at io.netty.util.NettyRuntime$AvailableProcessorsHolder.setAvailableProcessors(NettyRuntime.java:51)
[error]     at io.netty.util.NettyRuntime.setAvailableProcessors(NettyRuntime.java:87)
[error]     at org.elasticsearch.transport.netty4.Netty4Utils.setAvailableProcessors(Netty4Utils.java:82)
[error]     at org.elasticsearch.transport.netty4.Netty4Transport.<init>(Netty4Transport.java:138)
[error]     at org.elasticsearch.transport.Netty4Plugin.lambda$getTransports$0(Netty4Plugin.java:93)
[error]     at org.elasticsearch.node.Node.<init>(Node.java:430)
[error]     at org.codelibs.elasticsearch.runner.node.ClusterRunnerNode.<init>(ClusterRunnerNode.java:17)
[error]     at org.codelibs.elasticsearch.runner.node.ClusterRunnerNode.<init>(ClusterRunnerNode.java:23)
[error]     at org.codelibs.elasticsearch.runner.ElasticsearchClusterRunner.build(ElasticsearchClusterRunner.java:355)
[error]     ... 50 more
[error] Test akka.stream.alpakka.elasticsearch.ElasticsearchTest failed: java.lang.NullPointerException: null, took 0.0 sec
[error]     at akka.stream.alpakka.elasticsearch.ElasticsearchTest.teardown(ElasticsearchTest.java:70)
[error]     ...
```

--- a/build.sbt
+++ b/build.sbt
@@ -93,6 +93,7 @@ lazy val elasticsearch = project
   .enablePlugins(AutomateHeaderPlugin)
   .settings(
     name := "akka-stream-alpakka-elasticsearch",
+    parallelExecution in ThisBuild := false,
     Dependencies.Elasticsearch
   )
 

--- a/build.sbt
+++ b/build.sbt
@@ -93,8 +93,9 @@ lazy val elasticsearch = project
   .enablePlugins(AutomateHeaderPlugin)
   .settings(
     name := "akka-stream-alpakka-elasticsearch",
-    parallelExecution in ThisBuild := false,
-    Dependencies.Elasticsearch
+    Dependencies.Elasticsearch,
+    // For elasticsearch-cluster-runner https://github.com/akka/alpakka/issues/479
+    parallelExecution in Test := false
   )
 
 lazy val files = project // The name file is taken by `sbt.file`!

--- a/elasticsearch/src/test/java/akka/stream/alpakka/elasticsearch/ElasticsearchTest.java
+++ b/elasticsearch/src/test/java/akka/stream/alpakka/elasticsearch/ElasticsearchTest.java
@@ -41,13 +41,13 @@ public class ElasticsearchTest {
   public static void setup() throws IOException {
     runner = new ElasticsearchClusterRunner();
     runner.build(ElasticsearchClusterRunner.newConfigs()
-        .baseHttpPort(9210)
-        .baseTransportPort(9310)
+        .baseHttpPort(9200)
+        .baseTransportPort(9300)
         .numOfNode(1));
     runner.ensureYellow();
 
     //#init-client
-    client = RestClient.builder(new HttpHost("localhost", 9211)).build();
+    client = RestClient.builder(new HttpHost("localhost", 9201)).build();
     //#init-client
 
     //#init-mat

--- a/elasticsearch/src/test/java/akka/stream/alpakka/elasticsearch/ElasticsearchTest.java
+++ b/elasticsearch/src/test/java/akka/stream/alpakka/elasticsearch/ElasticsearchTest.java
@@ -40,7 +40,10 @@ public class ElasticsearchTest {
   @BeforeClass
   public static void setup() throws IOException {
     runner = new ElasticsearchClusterRunner();
-    runner.build(ElasticsearchClusterRunner.newConfigs().baseHttpPort(9210).baseTransportPort(9310).numOfNode(1));
+    runner.build(ElasticsearchClusterRunner.newConfigs()
+        .baseHttpPort(9210)
+        .baseTransportPort(9310)
+        .numOfNode(1));
     runner.ensureYellow();
 
     //#init-client

--- a/elasticsearch/src/test/scala/akka/stream/alpakka/elasticsearch/ElasticsearchSpec.scala
+++ b/elasticsearch/src/test/scala/akka/stream/alpakka/elasticsearch/ElasticsearchSpec.scala
@@ -235,6 +235,4 @@ class ElasticsearchSpec extends WordSpec with Matchers with BeforeAndAfterAll {
     }
   }
 
-
-
 }

--- a/elasticsearch/src/test/scala/akka/stream/alpakka/elasticsearch/ElasticsearchSpec.scala
+++ b/elasticsearch/src/test/scala/akka/stream/alpakka/elasticsearch/ElasticsearchSpec.scala
@@ -39,7 +39,13 @@ class ElasticsearchSpec extends WordSpec with Matchers with BeforeAndAfterAll {
   //#define-class
 
   override def beforeAll() = {
-    runner.build(ElasticsearchClusterRunner.newConfigs().baseHttpPort(9200).baseTransportPort(9300).numOfNode(1))
+    runner.build(
+      ElasticsearchClusterRunner
+        .newConfigs()
+        .baseHttpPort(9200)
+        .baseTransportPort(9300)
+        .numOfNode(1)
+    )
     runner.ensureYellow()
 
     register("source", "Akka in Action")
@@ -228,5 +234,7 @@ class ElasticsearchSpec extends WordSpec with Matchers with BeforeAndAfterAll {
       )
     }
   }
+
+
 
 }

